server/app/services/question/types/QuestionDefinition.java

```diff
@@ -201,8 +202,31 @@ public final String getValidationPredicatesAsString() {
   /** Get the default validation predicates for this question type. */
   abstract ValidationPredicates getDefaultValidationPredicates();
 
-  /** Validate that all required fields are present and valid for the question. */
+  /**
+   * Overload of {@code validate()} that sets {@code previousDefinition} to empty.
+   *
+   * @see #validate(Optional)
+   */
   public final ImmutableSet<CiviFormError> validate() {
+    return validate(Optional.empty());
+  }
+
+  /**
+   * Validate that all required fields are present and valid for the question.
+   *
+   * @param previousDefinition the optional previous version of the {@link QuestionDefinition}
+   * @throws IllegalArgumentException if the previousDefinition is not of the same type as this
+   *     definition.
+   */
+  public final ImmutableSet<CiviFormError> validate(
```

Comment:
**suggestion (blocking):** mention in the comment why the previous version is optionally provided -- what behavior changes

Reply:
Done!

Reply:
Also added to the javadoc in the QuestionService